core/src/main/java/com/linecorp/armeria/server/ExceptionReportingServerErrorHandler.java

```diff
@@ -0,0 +1,138 @@
+import com.linecorp.armeria.server.logging.LoggingService;
+
+import io.micrometer.core.instrument.Meter.Id;
+import io.micrometer.core.instrument.Meter.Type;
+import io.micrometer.core.instrument.Tags;
+import io.micrometer.core.instrument.cumulative.CumulativeCounter;
+
+/**
+ * A {@link ServerErrorHandler} that wraps another {@link ServerErrorHandler}
+ * to periodically report the exceptions that were not logged by decorators such as
+ * {@link LoggingService}.
+ */
+class ExceptionReportingServerErrorHandler implements ServerErrorHandler, ServerListener {
+
+    private static final Logger logger = LoggerFactory.getLogger(ExceptionReportingServerErrorHandler.class);
+
+    private final ServerErrorHandler delegate;
+    private final Duration duration;
+    private final CumulativeCounter counter;
+    private long lastExceptionsCount;
+    private boolean started;
+    @Nullable
+    private Throwable thrownException;
+    @Nullable
+    private ScheduledFuture<?> reportingTaskFuture;
+
+    ExceptionReportingServerErrorHandler(ServerErrorHandler serverErrorHandler, Duration duration) {
+        assert !duration.isNegative() && !duration.isZero();
+        delegate = serverErrorHandler;
+        this.duration = duration;
+        counter = new CumulativeCounter(new Id("armeria.server.unhandledExceptions", Tags.empty(),
+                                               null, null, Type.COUNTER));
+    }
+
+    /**
+     * If it's required to report exceptions, increase the counter and store first thrown exception.
+     */
+    @Nullable
+    @Override
+    public HttpResponse onServiceException(ServiceRequestContext ctx, Throwable cause) {
+        if (ctx.shouldReportUnhandledExceptions()) {
+            if (thrownException == null) {
+                thrownException = cause;
+            }
+            counter.increment();
+        }
+        return delegate.onServiceException(ctx, cause);
+    }
+
+    @Nullable
+    @Override
+    public AggregatedHttpResponse renderStatus(ServiceConfig config, @Nullable RequestHeaders headers,
+                                               HttpStatus status, @Nullable String description,
+                                               @Nullable Throwable cause) {
+        return delegate.renderStatus(config, headers, status, description, cause);
+    }
+
+    @Override
+    public void serverStarting(Server server) throws Exception {}
+
+    @Override
+    public void serverStarted(Server server) throws Exception {
+        if (started) {
+            return;
+        }
+
+        started = true;
+        reportingTaskFuture = server.config().workerGroup().scheduleAtFixedRate(
+                this::reportException, duration.toMillis(), duration.toMillis(), TimeUnit.MILLISECONDS);
+        server.config().meterRegistry().more().counter(counter.getId().getName(), counter.getId().getTags(),
+                                                       counter, CumulativeCounter::count);
```

Comment:
This introduces unnecessary level of indirections. Could you start with `null` `Counter`, and then lazily register a cumulative counter here? `meterRegistry().counter(...)` simply creates and registers a cumulative counter.